### ISSUE TITLE
QuickAction "checkbox" selector for display favorites groups one by one

### ISSUE THREAD

**Issue by ThibaultPelloquin** [osmandapp/OsmAnd#14287]
Hi! 

Is it possible to add an option in the Show/Hide favorites option in QuickAction menu to chose the groups to display one by one. Possibly with a top title "display all/hide all" ?

Thank you for your answer :)

**Comment by vshcherb** [comment]
Yes we need to add option to Show/Hide Favorites to select 1 or several Favorite Groups. State of this switch could be determined by state of the first group in the selected list

**Comment by vshcherb** [comment]
1. We need to redesign Configure Map -> Favorites similar to *new* Tracks list, so User could treat each Favorite List is similar to a separate GPX.
2. We need to reunderstand use cases for adding / managing a Favorite from a Hidden group. In case new Group is created it's automatically visible on the map.

Following that paradigm we need to have Favorites Quick Action - Show / Hide very similar or exactly same as we do for Tracks.

**Comment by vshcherb** [comment]
Problem to think: if we have **ultimate** quick action to select 1 group and hide all others, what will be a revertive action? 

In theory we should revert to previous state and disable that group but in practice we need to store that state somehow and it could be overridden with another **ultimate** aciton.

**Comment by ThibaultPelloquin** [comment]
Hi!

1. Yes, it makes sense.
2. A simple solution could be : if you add a favorite on a hidden group, this one is automatically displayed after creation of new favorite. If user want to hide the group, he has to hide it again.

For the ultimate quick action, we can imagine a 3 steps action?
[hide all] -> [show last custom state] -> [show all] (-> [hide all])

**Comment by vshcherb** [comment]
After a discussion we think we can easily introduce quick action for a Group On/Off without changing the state of another Groups if it solves use case, it's 1 task.

With a feature to hide other groups, we think it requires a special state "highlight group(s)", so it doesn't matter the state of previous groups they will be **all off** and after click again they will be reverted to the same state. Unfortunately we can't introduce it only for quick action that state will need to be manageable "highlight" in Configure Map / My Places to avoid confusion. And in general it makes this task quite complicated. 

Another side story we could introduce... context menu for Favorite Groups.
